### ISSUE TITLE
Implement lumberjack protocol for input/sink

### ISSUE THREAD

**Issue by ict-one-nl** [opensearch-project/data-prepper#3333]
**Is your feature request related to a problem? Please describe.**
The forward protocol is used for filebeat/logstash setups to forward logs between instances. This is very useful for larger architectures. It would be nice to implement this for data prepper as well so data prepper can be used as an intermediate or endpoint before sending the data to Opensearch and can fully replace logstash.

**Describe the solution you'd like**
Implement input/sink

**Describe alternatives you've considered (Optional)**
http - overhead might be an issue

**Additional context**
none

**Comment by dlvenable** [comment]
@ict-one-nl , I think this is a duplicate of #950. Is there anything different in your request?

**Comment by ict-one-nl** [comment]
Yeah, although the original post doesn't reference lumberjack by name :)
